lib/msf/base/sessions/command_shell.rb

```diff
@@ -312,18 +312,17 @@ def cmd_shell(*args)
   end
 
   #
-  # Check if there is a binary in PATH env
+  # Returns path of a binary in PATH env.
   #
   def binary_exists(binary)
     print_status("Trying to find binary(#{binary}) on target machine")
-    binary_path = shell_command_token("which #{binary}").to_s.strip
-    if binary_path.eql?("#{binary} not found") || binary_path == ""
-      print_error(binary_path)
+    binary_path = shell_command_token("command -v '#{binary}' || echo 'false'")
+    if binary_path.to_s.include?('false')
```

Comment:
Just confirming; Is it possible to have false matches here if the result happens to have `false` in it? 👀

Reply:
Yeah right, I tried with `/tmp/false/something` it was not able to detect the binary exists or not. Will change it to `&& echo true`.